Source/WebKit/UIProcess/WebsiteData/WebsiteDataStore.cpp

```diff
@@ -300,7 +300,7 @@ void WebsiteDataStore::resolveDirectoriesIfNecessary()
         m_resolvedConfiguration->setResourceLoadStatisticsDirectory(resolveAndCreateReadWriteDirectoryForSandboxExtension(m_configuration->resourceLoadStatisticsDirectory()));
     if (!m_configuration->privateClickMeasurementStorageDirectory().isEmpty())
         m_resolvedConfiguration->setPrivateClickMeasurementStorageDirectory(resolveAndCreateReadWriteDirectoryForSandboxExtension(m_configuration->privateClickMeasurementStorageDirectory()));
-    if (!m_configuration->serviceWorkerRegistrationDirectory().isEmpty() && m_resolvedConfiguration->serviceWorkerRegistrationDirectory().isEmpty())
```

Comment:
Below, there are a few that make sure the destination is empty to avoid overwriting. Why is this right here but not below?

Reply:
It's wrong below; I will update.

Reply:
(m_resolvedConfiguration is initialized to be m_configuration, so there's no point checking if its directory is empty.)